### PR TITLE
IN-49 Vuln Sync CSV Upload Use Case

### PR DESCRIPTION
**Pull request by ax-jmiller**
# Description

A new use case to show automation of the Vulnerability Sync ingest process. I added docstrings to make this script more PEP compliant, hopefully this aids the consumers of this script in understanding how it works.

## Motivation and Context

Fixes # (issue)

### Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have updated the documentation accordingly
- [x] I have tested the changes

### CODE REVIEW

File: examples/use-cases/vuln_sync_csv_upload/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+```
+Sending vuln.csv to Automox Vulnerability Sync...
+==============================
+BATCH ID: 501
+vuln.csv has been uploaded.
+==============================
+Moving vuln.csv to /automox-console-sdk-python/examples/use-cases/vuln_sync_csv_upload/cve_queue/processed/vuln.csv
+```
+
+Once the script has iterated over all of the elgibile files for upload, it will poll the API for the build status of the batches sent, and when all the batches are marked as "Awaiting Approval", a pre-task report for each batch will be outputted.
```
Comment (ax-tschmidtke) [inline examples/use-cases/vuln_sync_csv_upload/README.md:45]: typo

File: examples/use-cases/vuln_sync_csv_upload/vuln_upload.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+import requests
+
+
+def upload_cve(file: FileIO):
+    """ Uploads vulnerability list to Automox Vulnerability Sync endpoint.
+
+    Args:
+        file (FileIO): A CSV file containing vulnerability data.
+
+    Returns:
```
Comment (ax-tschmidtke) [inline examples/use-cases/vuln_sync_csv_upload/vuln_upload.py:17]: You can annotate that the response is a dict like this: `def upload_cve(file: FileIO) -> dict`, helpful for IntelliSense and other introspection.

File: examples/use-cases/vuln_sync_csv_upload/vuln_upload.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        response_data (dict): API response from Automox Vulnerability Sync
+        https://developer.automox.com/openapi/vulnsync/operation/UploadCSVBatch/
+    """
+
+    response_data = {}
+
+    task = "patch"
+    url  = f"https://console.automox.com/api/orgs/{organization}/tasks/{task}/batches/upload"
```
Comment (ax-tschmidtke) [inline examples/use-cases/vuln_sync_csv_upload/vuln_upload.py:25]: Wasn't there something where the API endpoints needed you to also specify the `o=<org_id>` query parameter or was that changed?

File: examples/use-cases/vuln_sync_csv_upload/vuln_upload.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+            msg = ""
+            msg = msg.join(response_data['errors'])
+
+            raise Exception(msg)
+    except requests.RequestException as error:
+        print(f"Error: Unable to complete CSV upload request. ({error})")
+
+    return response_data
+
+def get_unprocessed_cves(directory: str):
```
Comment (ax-tschmidtke) [inline examples/use-cases/vuln_sync_csv_upload/vuln_upload.py:52]: Same as above `() -> list`

File: examples/use-cases/vuln_sync_csv_upload/vuln_upload.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    Args:
+        directory (str): Directory to look in for CSVs.
+
+    Returns:
+        cve_files (list): List of files to be processed and uploaded.
+    """
+
+    cve_files = []
+
+    paths = os.listdir(directory)
```
Comment (ax-tschmidtke) [inline examples/use-cases/vuln_sync_csv_upload/vuln_upload.py:64]: You might be able to clean up the loop a bit by using `glob`: https://docs.python.org/3/library/glob.html

File: examples/use-cases/vuln_sync_csv_upload/vuln_upload.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+
+    Args:
+        uploaded_batches (dict): A dictionary of the latest responses from the Automox API about the status of a batch.
+
+    Returns:
+        uploaded_batches: An updated dictionary of the latest responses from the Automox API about the status of a batch.
+    """
+
+    for batch_id, batch in uploaded_batches.items():
+        if uploaded_batches[batch_id]['status'] != "awaiting_approval":
```
Comment (ax-tschmidtke) [inline examples/use-cases/vuln_sync_csv_upload/vuln_upload.py:136]: You could also do `batch['status']` here.

File: examples/use-cases/vuln_sync_csv_upload/vuln_upload.py [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+            f"Unknown Hosts: {batch['unknown_host_count']}\n"
+            "=============================="
+        )
+
+        print(output)
+except Exception as e:
+    print(f"Error: {e}\n")
+    raise
+except KeyboardInterrupt:
+    print ("Crtl+C Pressed. Shutting down.")
```
Comment (ax-tschmidtke) [inline examples/use-cases/vuln_sync_csv_upload/vuln_upload.py:197]: typo

**Review by ax-tschmidtke — CHANGES_REQUESTED** [review]
Looks great, just a few small things!

File: examples/use-cases/vuln_sync_csv_upload/vuln_upload.py (hunk 3 above)
Comment (ax-jmiller) [reply examples/use-cases/vuln_sync_csv_upload/vuln_upload.py:25]: I don't think so.

The docs only reference path params: https://developer.automox.com/openapi/vulnsync/operation/UploadCSVBatch/

Comment (ax-tschmidtke) [reply examples/use-cases/vuln_sync_csv_upload/vuln_upload.py:25]: 👍

**Review by ax-tschmidtke — APPROVED** [review]
:shipit:

**Pull request merged**